react/README.md

```diff
@@ -498,22 +450,76 @@ This guide provides a few guidelines on writing sensible React. Many of these ru
   <Good status="nice and clean" />
   ```
 
-- [5.6](#5.6) <a name="5.6"></a> Name props that act as event listeners `on<EventName`.
+- [5.6](#5.6) <a name="5.6"></a> Name props that act as event listeners `on<EventName>`, and methods that are called in response to events `handle<EventName>`.
 
   ```js
   // bad
-  class BadComponent extends React.Component {
-    static propTypes = {click: React.PropTypes.func}
+  class BadComponent extends Component {
+    static propTypes = {click: PropTypes.func}
   }
 
-  <BadComponent click={() => console.log('clicked')} />
+  class BadComponentUser extends Component {
+    _onClick() {
+      console.log('clicked');
+    }
+
+    render() {
+      return <BadComponent click={this._onClick} />
+    }
+  }
 
   // good
-  class GoodComponent extends React.Component {
-    static propTypes = {onClick: React.PropTypes.func}
+  class GoodComponent extends Component {
+    static propTypes = {onClick: PropTypes.func}
   }
 
-  <GoodComponent onClick={() => console.log('clicked')} />
+  class GoodComponentUser extends Component {
+    handleClick() {
+      console.log('clicked');
+    }
+
+    render() {
+      return <GoodComponent onClick={this.handleClick} />
+    }
+  }
+  ```
+
+- [5.7](#5.7) <a name="5.7"></a> Avoid passing bound methods or arrow functions in the props of rendered components. Bind these in the constructor or use class properties instead.
+
+  > Why? Passing a bound function or using an arrow function creates a new function for each render, which increases memory usage and prevents using shallow rendering on subcomponents.
+
+  ESLint rules: [`jsx-no-bind`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md)
+
+  ```js
+  // bad
+  class BadComponent extends Component {
+    render() {
+      return <button onClick={this.handleClick.bind(this)} onMouseEnter={() => this.handleMouseEnter()} />;
+    }
+  }
+
+  // good
+  class GoodComponent extends Component {
+    constructor(...args) {
+      super(...args);
+      this.handleClick = this.handleClick.bind(this);
+      this.handleMouseEnter = this.handleMouseEnter.bind(this);
+    }
+
+    render() {
+      return <button onClick={this.handleClick} onMouseEnter={this.handleMouseEnter} />;
+    }
+  }
+
+  // also good
+  class GoodComponent extends Component {
+    handleClick = this.handleClick.bind(this);
+    handleMouseEnter = this.handleMouseEnter.bind(this);
+
+    render() {
+      return <button onClick={this.handleClick} onMouseEnter={this.handleMouseEnter} />;
+    }
```

Comment:
@perrupa nope, the properties above this are class instance properties, so they are still accessed with `this` (they just give you a quicker way of setting up instance properties that are constant expressions than to need to do it in the constructor).